sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -399,7 +402,16 @@ class AstBuilder extends SqlBaseBaseVisitor[AnyRef] with Logging {
    * separated) relations here, these get converted into a single plan by condition-less inner join.
    */
   override def visitFromClause(ctx: FromClauseContext): LogicalPlan = withOrigin(ctx) {
-    val from = ctx.relation.asScala.map(plan).reduceLeft(Join(_, _, Inner, None))
+    val relationPlan = ctx.relation.asScala.map {
+      case jrc: JoinRelationContext if jrc.broadcast != null =>
+        if (jrc.broadcast.left == null) { // default
+          Hint("BROADCAST_RIGHT", Seq.empty[String], plan(jrc))
```

Comment:
It does.

However here is my counter argument :)... In this case we had to define the syntax for a hint in the grammar (join will break without it). If we have to do that we are tied to that particular hint anyway, so in those cases I would implement it right away. The added bonus here would be that you simplify your analysis rule by quite a bit.